File: problemas/iniciante/1158/1158.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

let somaPA = (a0, n) => {
    a0 += 1 - (Math.abs(a0) % 2);
    let an = a0 + (n - 1) * 2;

    return Math.floor(((a0 + an) * n) / 2);
};

let N = parseInt(lines.shift().trim());
for (let i = 0; i < N; ++i) {
    let [X, Y] = lines.shift().trim().split(' ').map((x) => parseInt(x));

    console.log(somaPA(X, Y));
}

File: problemas/iniciante/1158/1158.go
package main

import (
    "fmt"
    "math"
)

func somaPA(a0 int, an int, n int) int {
    return (a0 + an) * n / 2
}

func main() {
    var n, x, y int

    fmt.Scanf("%d", &n)
    for i := 0; i < n; i++ {
        fmt.Scanf("%d %d", &x, &y)

        x += 1 - (int(math.Abs(float64(x))) % 2)

        fmt.Printf("%d\n", somaPA(x, x+2*(y-1), y))
    }
}
